Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/EntryDelegate.cpp

```diff
@@ -148,7 +154,163 @@ namespace AzToolsFramework
+                        else if (indexBellow.isValid() && aboveSourceEntry) // The index above is a source entry
+                        {
+                            DrawBranchPixMap(EntryBranchType::Last, painter, branchIconTopLeft, iconSize); // Draw First child Icon
+                        }
+                        else if (indexAbove.isValid() && bellowSourceEntry) // The index bellow is a source entry
+                        {
+                            DrawBranchPixMap(EntryBranchType::First, painter, branchIconTopLeft, iconSize); // Draw Last Child Icon
+                        }
+                        else //the index above and bellow are also child entries
+                        {
+                            DrawBranchPixMap(EntryBranchType::Middle, painter, branchIconTopLeft, iconSize); // Draw Default child Icon.
+                        }
+                    }
+
+                    remainingRect.adjust(thumbX, 0, 0, 0); // bump it to the right by the size of the thumbnail
+                    remainingRect.adjust(ENTRY_SPACING_LEFT_PIXELS, 0, 0, 0); // bump it to the right by the spacing.
+                }
+                QString displayString = index.column() == aznumeric_cast<int>(AssetBrowserEntry::Column::Name)
+                    ? qvariant_cast<QString>(entry->data(aznumeric_cast<int>(AssetBrowserEntry::Column::Name)))
+                    : qvariant_cast<QString>(entry->data(aznumeric_cast<int>(AssetBrowserEntry::Column::Path)));
+
+                style->drawItemText(
+                    painter, remainingRect, optionCopy.displayAlignment, actualPalette, isEnabled, displayString,
+                    isSelected ? QPalette::HighlightedText : QPalette::Text);
+            }
+        }
+
+        void SearchEntryDelegate::DrawBranchPixMap(
+            EntryBranchType branchType, QPainter* painter, const QPoint& point, const QSize& size) const
+        {
+            QPixmap pixmap;
+            AZ::IO::BasicPath<AZ::IO::FixedMaxPathString> absoluteIconPath;
+            switch (branchType)
+            {
+            case AzToolsFramework::AssetBrowser::EntryBranchType::First:
+                absoluteIconPath = AZ::IO::FixedMaxPath(AZ::Utils::GetEnginePath()) / TreeIconPathFirst;
+                break;
+            case AzToolsFramework::AssetBrowser::EntryBranchType::Middle:
+                absoluteIconPath = AZ::IO::FixedMaxPath(AZ::Utils::GetEnginePath()) / TreeIconPathMiddle;
+                break;
+            case AzToolsFramework::AssetBrowser::EntryBranchType::Last:
+                absoluteIconPath = AZ::IO::FixedMaxPath(AZ::Utils::GetEnginePath()) / TreeIconPathLast;
+                break;
+            case AzToolsFramework::AssetBrowser::EntryBranchType::OneChild:
+            default:
+                absoluteIconPath = AZ::IO::FixedMaxPath(AZ::Utils::GetEnginePath()) / TreeIconPathOneChild;
+                break;
+            }
+
+            pixmap.load(absoluteIconPath.c_str());
```

Comment:
question: This might just be the way things work in Qt, but do you have to load this every time you draw? This seems like it could potentially be slow, is there a better alternative (keeping a reference to the loaded pixmap around?)

Reply:
Yeah that sounds like the best option. Gonna explore that path